docs/protocol/core/v3.0.rst

```diff
@@ -1228,56 +1192,22 @@ operations:
     with a trailing slash ``/`` and store can assume there is as least one key
     that stars with prefix.
 
-
 ``list_dir`` - Retrieve all keys and prefixes with a given prefix and
 which do not contain the character "/" after the given prefix.
 
-    | Parameters: `prefix`, ends with a trailing slash ``/``
+    | Parameters: `prefix`
     | Output: set of `keys` and set of `prefixes`
 
     For example, if a store contains the keys "a/b", "a/c", "a/d/e",
     "a/f/g", then ``list_dir("a/")`` would return keys "a/b" and "a/c"
-    and prefixes "a/d/" and "a/f/".
-
-    On non-existing prefix, store may return the empty set.
-
+    and prefixes "a/d/" and "a/f/". ``list_dir("b/")`` would return
+    the empty set.
 
-    Note: The requirement on trailing slashes is to avoid
-    search returning keys in the same hierarchy level but longer name, and
-    potentially expensive logic testing for the present of trailing slash on
-    each query. e.g:
 
-     - /meta/foo
-     - /meta/foo/dataset
-     - /meta/foobar
-
-     list_dir('/meta/foo') == '/meta/foo'&'/meta/foobar'
-     list_dir('/meta/foo/') == '/meta/foo/dataset'
-
-
-    Stores Must return trailing slashes in key responses when those
-    are prefix of other keys.
-
-    Like would ``list_dir('/meta/mydir')`` returns:
-       - ``/meta/path1``
-       - ``/meta/path2``
-       - ``/meta/path3/``
-       - ``/meta/path4/``
-
-    Thus we know that ``path1``, and ``path2`` are terminal objects with data,
-    and that ``/meta/path3`` and ``/meta/path4``.
-
-
-    This is similar to ``ls -p`` on Unix systems.
-
-    Note: In practice this means that this means most returned keys always ends in
-    ``/``, ``.json``, ``.array``, ``.group``, they will otherwise be chunks
-    data.
```

Comment:
I removed this for the sake of simplicity. I think we can just say, by definition, a prefix is a string with a trailing slash. The ``list_dir`` operation accepts a prefix.